pkg/storages/s3/session.go

```diff
@@ -231,9 +232,53 @@ func createSession(bucket string, settings map[string]string) (*session.Session,
 			}
 		})
 	}
+
+	if encodedHeaders, ok := settings[RequestAdditionalHeaders]; ok {
+		headers, err := getHeaders(encodedHeaders)
+		if err != nil {
+			return nil, err
+		}
+
+		s.Handlers.Validate.PushBack(func(request *request.Request) {
+			for k, v := range headers {
+				request.HTTPRequest.Header.Add(k.(string), v.(string))
+			}
+		})
+	}
+
 	return s, err
 }
 
+func getHeaders(encodedHeaders string) (map[interface{}]interface{}, error) {
+	var data interface{}
+	err := yaml.Unmarshal([]byte(encodedHeaders), &data)
+	if err != nil {
+		return nil, NewFolderError(err, "Invalid %s setting", RequestAdditionalHeaders)
+	}
+
+	headers, ok := data.(map[interface{}]interface{})
+	if !ok {
+		headerList, ok := data.([]interface{})
+		if !ok {
+			return nil, NewFolderError(err, "Invalid %s setting", RequestAdditionalHeaders)
+		}
+		headers = refomHeaderListToMap(headerList)
+	}
+
+	return headers, nil
+}
+
+func refomHeaderListToMap(headerList []interface{}) map[interface{}]interface{} {
```

Comment:
```suggestion
func refomHeaderListToMap(headerList []interface{}) map[string]string {
```
Might as well put the types as strings if only use is going to be casting all to string